File: ROS/day2/hw1/custom_interfaces/src/sub.cpp

#include <chrono>
#include <functional>
#include <memory>
#include <string>


#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "std_msgs/msg/int64.hpp"
#include "std_msgs/msg/float64.hpp"

class Sub : public rclcpp::Node
{
    public:
       Sub() : Node("sub")
        {
                subscriber = this ->create_subscription<std_msgs::msg::Int64>(
                "topicname", 
                10,
                std::bind(&Sub::topic_callback1, this, std::placeholders::_1));
        }

    private:
       void topic_callback1(const std_msgs::msg::Int64::SharedPtr msg1)
       {
           RCLCPP_INFO(this->get_logger(), "Received message: '%ld'", msg1->data);
       }
       rclcpp::Subscription<std_msgs::msg::Int64>::SharedPtr subscriber;

};


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<Sub>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ROS/day1/hw2/hw2_package/my_first_ros_rclpy_pkg/my_first_ros_rclpy_pkg/sub.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String
from std_msgs.msg import Int32
from std_msgs.msg import Float64


class HelloworldSubscriber(Node):
    def __init__(self):
        super().__init__('helloworld_subscriber')
        self.helloworld_subscriber1 = self.create_subscription(
            String, 
            'topicname', 
            self.subscribe_topic_message,
            10)
        self.helloworld_subscriber2 = self.create_subscription(
            Int32, 
            'topicname_int', 
            self.subscribe_topic_message,
            30)
        self.helloworld_subscriber3 = self.create_subscription(
            Float64, 
            'topicname_float', 
            self.subscribe_topic_message,
            30)

    def subscribe_topic_message(self, msg):
        self.get_logger().info('Received message: {0}'.format(msg.data))



def main(args=None):
    rclpy.init(args=args)
    node = HelloworldSubscriber()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()